test/e2e/framework/create.go

```diff
@@ -655,3 +658,32 @@ func PrettyPrint(item interface{}) string {
 	}
 	return fmt.Sprintf("%+v", item)
 }
+
+type SecretFactory struct{}
+
+func (f *SecretFactory) New() runtime.Object {
+	return &v1.Secret{}
+}
+
+func (*SecretFactory) Create(f *Framework, i interface{}) (func() error, error) {
+	item, ok := i.(*v1.Secret)
+	if !ok {
+		return nil, ItemNotSupported
+	}
+
+	client := f.ClientSet.CoreV1().Secrets(f.Namespace.GetName())
+	if _, err := client.Create(item); err != nil {
+		return nil, errors.Wrap(err, "create Secret")
+	}
+	return func() error {
+		return client.Delete(item.GetName(), &metav1.DeleteOptions{})
+	}, nil
+}
+
+func (*SecretFactory) UniqueName(i interface{}) string {
```

Comment:
MetaNamespaceKeyFunc?

https://godoc.org/k8s.io/client-go/tools/cache#MetaNamespaceKeyFunc

Reply:
Thanks for the pointer. MetaNamespaceKeyFunc is not quite the same (UniqueName will ignore the namespace of items which are known to be not namespaced), but as this shouldn't matter in practice (namespace shouldn't be set then), I'll remove all of the UniqueName methods with it.